### DISCUSSION TITLE
[deployment] Select a CSP and prepare for production situations

### DISCUSSION THREAD

**Discussion by frgfm** [pyronear/pyro-api#194] [General]
Currently the app is hosted freely but with high constraints on up time, this needs to change when users require constant access. A few things need to be taken care of:
- [x] Benchmark & select CSP
- [ ] Integrate a service module to use it with the API

**Comment by chloeskt** [comment]
There is [Outscale](https://fr.outscale.com/) it's a French company (cloud computing + cloud service). However it was created by Dassault System to ensure french sovereignty. 

To know more what their engagements: https://fr.outscale.com/rse-nous-sommes-engages-et-responsables/

For the prices: https://fr.outscale.com/tarifs/

**Comment by jean-pasquier** [comment]
I had a look to [outscale](https://fr.outscale.com/machines-virtuelles/) and it seams to be an IaaS (Infrastructure as a Service) is it what we are looking for ?

https://www.redhat.com/fr/topics/cloud-computing/iaas-vs-paas-vs-saas
TLDR:
* IaaS = we need to manage installation, deployment, etc. on a VM (eg AWS EC2, etc.)
* PaaS = deployment managed (eg Heroku)

I found a PaaS using Ouscale infrastructures named [Scalingo](https://fr.outscale.com/communiques-presse/scalingo-sappuie-sur-le-cloud-de-3ds-outscale-pour-offrir-fiabilite-et-securite/) having a good price/performance ratio based on [this French PaaS benchmark](https://devops-life.com/blog/2019/01/24/clevercloud-vs-scalingo/) - 0,01 € / hour for the smallest config and a free Postres plan, according to its [pricing](https://scalingo.com/fr/pricing/osc-fr1) what do you think ?

**Comment by frgfm** [comment]
Thanks for the investigation!
I would imagine that IaaS are thus cheaper to PaaS for comparable machines?
If so, considering our simple setup, it could be manageable. But so we have two aspects to compare: pricing/performance, and business model of the company (checking there is no misalignment with Pyronear values as much as possible). Any idea where those CSP stand regarding that last aspect?

**Comment by florianriche** [comment]
Quick note: The choice of CSP is quite impacting on other parts of our system (e.g pyro-riks would love a platform that enable jobs to be launched automatically or have a Reddis Cache). The CSP selection should be advertised and/or discussed with other teams.

As a side note, should there be a place to gather global architecture discussions? I have the feeling that we have general impacting talks scattered among the different repos.

**Comment by frgfm** [comment]
Not really on Github, so perhaps we should schedule a meeting for this very topic?

**Comment by chloeskt** [comment]
Yes I agree with @frgfm we should not discuss this on GitHub, let’s setup a call 🙂

**Comment by chloeskt** [comment]
Would it be too complicated to manage VMs on our own ? 

I've had a quick look at some PaaS:
- OVH (guess that it's well-known)
- [Orange Business Services](https://www.orange-business.com/fr/produits/gamme-cloud-computing-infrastructures-it)
- [platform.sh](https://platform.sh/product/)
- [gandi](https://www.gandi.net/fr/simple-hosting). They support some nice projects ([here](https://www.gandi.net/fr/gandi-supports)) such as WWF, fidh, etc. They also have a nice "index de l'égalité salariale Femmes-Hommes" of 87/100. 

It did not benchmark yet the prices of all the solutions mentioned above, but so far `gandi` seems to be the closest to Pyronear's values.

**Comment by frgfm** [comment]
My bad, I was supposed to take care of this over the weekend. I'll start the deployment on OVH to try to have something working for Wednesday!

**Comment by frgfm** [comment]
FYI: we're in the middle of deployment on an OVH sandbox :ok_hand:

**Comment by frgfm** [comment]
I was about to close this issue because we now have a deployed instance on OVH, but I wanted first to suggest another option that is surprisingly free: Deta

It's suggested by FastAPI and seems quite modern : https://fastapi.tiangolo.com/tr/deployment/deta/

**Comment by florianriche** [comment]
Hi, 
It seems like a good solution indeed that would have fit our usecase but why would we switch now that we have a solution on OVH?

**Comment by frgfm** [comment]
> Hi,
> It seems like a good solution indeed that would have fit our usecase but why would we switch now that we have a solution on OVH?

Quite simple: it's free :sweat_smile:

**Comment by florianriche** [comment]
Oh my bad I thought we were in a free tier plan on OVH. 
I would agree to do the switch then :)

**Comment by frgfm** [comment]
Moving this to a question since it will remain an ongoing matter as new providers appear :)
